### ISSUE TITLE
Add custom refresh / scan rate

### ISSUE THREAD

**Issue by JonnyTech** [usr-sse2/RDM#5]
RDM allows one to add a custom **horizontal** and **vertical** resolution, but it does not allow adding a custom **refresh** resolution.

My monitor supports refresh of 50Hz and 60Hz but RDM only shows 60Hz.

[SwitchResX](https://madrau.com/srx_howtouseit/srx_howtouseit/srx_timings.html) allows adding a custom resolution with  **horizontal** and **vertical** and **refresh** values Shown as `Scan rate` in the image below:

![image](https://user-images.githubusercontent.com/3584732/51554535-6bc74200-1e6d-11e9-8f23-3ff0a780a2a3.png)

After rebooting the new 50Hz option is available in RDM menus too.

Can this feature be added to `RDM` too? It is the only thing that I find lacking in this excellent app!

**Comment by usr-sse2** [comment]
The current implementation can add only scaled resolutions. It means that the GPU renders the image in the specified resolution and then up- or downscales it to the native resolution of the display. The actual display mode is not changed. Adding scaled resolutions is achieved by simply adding them into a plist.
SwitchResX can also add new output modes (resolution + refresh rate), which is achieved by editing the display EDID and adding EDID override to the plist. It is a much more complex task with many variable parameters and algorithms of generating. I have no plans of implementing it now.

**Comment by JonnyTech** [comment]
> achieved by editing the display EDID and adding EDID override to the plist

Thanks for your reply @usr-sse2 - do you have any documentation on the process?

> I have no plans of implementing it now

Maybe in the future?

**Comment by usr-sse2** [comment]
Here you can read (in Russian) how to obtain your monitor's EDID and open it with AW EDID Editor. 
https://osxpc.ru/zavod/notebook/lvds-prefpane/
Here you can change even more than with SwitchResX. Then go to the display override file for your monitor in `/System/Library/Displays/Contents/Resources/Overrides` and add `IODisplayEDID` key containing your modified EDID data.

Note: I didn't test this way.

**Comment by JonnyTech** [comment]
Thank you, I shall investigate.

**Comment by usr-sse2** [comment]
I patched EDID using AW EDID Editor and put it into the override file, it works. Added 50 Hz refresh rate and removed interlaced modes for Lenovo P24q display.

**Comment by JonnyTech** [comment]
Yes, indeed, the process works, but is tedious especially for non-techies.

I am trying to use [edid-decode](https://git.linuxtv.org/edid-decode.git) to automate - with help from https://gist.github.com/ejdyksen/8302862 - not sure if possible to do though..

**Comment by ayechief** [comment]
This is one of the reasons I hate Macs... I cannot even set 4k 75Hz on 2k 75Hz display where on Windows I do it without any trouble, just by simply selecting Superresolution (or something like this) in AMD GPU and then in Windows settings I just set it to 4k and I can set 75 Hz.
The issue on Macs is I can't just set whatever resolution I want with 75 Hz but 2k... This hurts me so bad because I can't use so little UI with so little text.

**Comment by radianttap** [comment]
It's an old issue but I wonder where is RDM getting available refresh rates? For LG 27GN950 I see 60Hz and 95Hz only while the monitor is capable of 120Hz and 144Hz (even 160Hz when overclocked).

Here's the full EDID info extracted. Extension block contains entries for both 95Hz and 144Hz thus I'm not sure why RDM only shows 60/95? BTW, SwitchResX also offers only 95Hz. It's puzzling.

My GPU is RX 5500 XT, connected over DisplayPort 1.4 DSC. 

[27GN950-EDID.txt](https://github.com/usr-sse2/RDM/files/6220867/27GN950-EDID.txt)
